Tests/MapboxMapsTests/Location/LocationManagerTests.swift

```diff
@@ -37,4 +37,27 @@ internal class LocationManagerTests: XCTestCase {
         XCTAssertNil(locationManager.delegate)
         XCTAssertFalse(locationManager.showUserLocation)
     }
+
+    func testLocationManagerPuckUpdate() {
+        var locationOptions = LocationOptions()
+        locationOptions.showUserLocation = true
+        let locationManager = LocationManager(locationOptions: locationOptions,
+                                              locationSupportableMapView: locationSupportableMapMock)
+
+        var locationOptions2 = LocationOptions()
+        locationOptions2.puckType = .puck2D(Puck2DConfiguration(scale: .constant(2.0)))
+        locationOptions2.showUserLocation = true
```

Comment:
@nishant-karajgikar @julianrex it occurs to me that puckType is only relevant if showUserLocation is true. This suggests to me that we are modeling this inefficiently. What if we added a `.none` to the `PuckType` enum instead? (not as part of this PR — we'd ticket it separately).

cc: @tobrun

Reply:
@nishant-karajgikar @julianrex @tobrun here's the proposed change: https://github.com/mapbox/mapbox-maps-ios/pull/203